src/librbd/cache/pwl/rwl/PmemManager.cc

```diff
@@ -0,0 +1,192 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "PmemManager.h"
+
+#undef dout_subsys
+#define dout_subsys ceph_subsys_rbd_pwl
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::cache::pwl::rwl::PmemManager: " \
+                           << this << " " <<  __func__ << ": "
+
+namespace librbd::cache::pwl::rwl {
+
+/* p2align and p2roundup require uint64_t, keep the same with offset */
+static const uint64_t min_write_alloc_size = MIN_WRITE_ALLOC_SIZE;
+
+PmemDev::PmemDev(const char * path, const uint64_t size, CephContext *cct)
+  : m_lock(ceph::make_mutex(pwl::unique_lock_name(
+      "librbd::cache::pwl::PmemManager::m_lock", this))),
+    m_cct(cct)
+{
+  m_head_addr = create_dev(path, size);
+  if (!m_head_addr) {
+    lderr(m_cct) << "failed to create pmem device." << dendl;
+    return;
+  }
+  m_max_offset = p2align(m_mapped_len, min_write_alloc_size);
+  ldout(m_cct, 5) << "successfully created pmem Device."
+                  << " m_is_pmem: " << m_is_pmem
+                  << " m_head_addr: " << (void *)m_head_addr
```

Comment:
OK, I will revise it later. Because I want to ask another related question. I have used a lot of C-style cast in my code, So does original code without my PR, such as:
```
m_pmem_log_entries = (struct WriteLogCacheEntry *)((char *)m_superblock +
                         FIRST_ENTRY_OFFSET); 
head_addr = (char *)pmem_map_file(...);
i.copy((unsigned)log_entry->write_bytes(), (char*)log_entry->cache_buffer);
```
Should I also change them all to `static_cast`?